src/main/java/core/basesyntax/strategy/impl/PurchaseOperationImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationHandler;
+
+public class PurchaseOperationImpl implements OperationHandler {
+    @Override
+    public void applyOperation(FruitTransaction transaction) {
+        if (transaction.getQuantity() <= MIN_QUANTITY) {
+            throw new RuntimeException("Purchase must be 1 or more items. But was: "
+                    + transaction.getQuantity());
+        }
+        if ((getCurrentBalance(transaction) - transaction.getQuantity()) < 0) {
+            throw new RuntimeException("There is no required quantity in stock.Exist: "
+                    + getCurrentBalance(transaction) + "Required: "
+                    + transaction.getQuantity());
```

Comment:
The error message in the RuntimeException is misleading. It says 'There is no required quantity in stock.Exist: ' but does not include a space between 'stock.' and 'Exist:'. Also, 'Required:' should be followed by a space for better readability.